lib/led-matrix/src/LEDMatrix.cpp

```diff
@@ -0,0 +1,62 @@
+/* LED Matrix Source File */
+
+#include "LEDMatrix.h"
+
+// Constructor: An LED Matrix is defined by the pins it is connected to.
+LEDMatrix ::LEDMatrix(PinName R, PinName G, PinName B) : _RChannel(R), _GChannel(G), _BChannel(B) {}
+
+// Take in values from 0 to 255 for each color stream and maps them to a PWM signal.
+LEDMatrix ::void setColorRGB(uint8_t R, uint8_t G, uint8_t B) {
+  _RChannel.pulsewidth((1.0 / 255.0) * R);
+  _GChannel.pulsewidth((1.0 / 255.0) * G);
+  _BChannel.pulsewidth((1.0 / 255.0) * B);
+}
+
+// Just for convenience. Call setColorRGB().
+LEDMatrix ::void setColor(color c) {
+  switch (c) {
+    case color::RED:
+      setColorRGB(255, 0, 0);
+      break;
+    case color::GREEN:
+      setColorRGB(0, 255, 0);
+      break;
+    case color::BLUE:
+      setColorRGB(0, 0, 255);
+      break;
+  }
+}
+
+// Set the specified color, wait, turn off all LEDs, wait, and repeat until another function is called by the main.cpp.
+LEDMatrix ::void setFlashingColorRGB(uint8_t R, uint8_t G, uint8_t B, uint8_t frequency_hz) {
+  float flash_num  = frequency_hz * HW_BRIDGE::LEDMATRIX::NUM_FLASHES;  // Flash for 10sec
+  float timeout_ms = (1.0 / frequency_hz) * 1000;
```

Comment:
rather than having an 8-bit integer frequency, you can just have a floating point frequency imo.

Reply:
On top of this, it would be really nice if we could use chrono here. Maybe like a `flash_period_duration` or smth (recipricol of freq).

Reply:
I think since we use chrono extensively, but deal a lot with frequencies instead of duration, we might benefit from a frequency convience class(internally using chrono) that does translations back and forth. 

Could do smth like https://stackoverflow.com/questions/36043351/stdchrono-for-frequencies. Maybe leave that as a new issue and just maybe switch to chrono types here?

Thoughts @younesr1 ?

Reply:
yeah i agree. instead of taking in a variable called frequency that is a uint8, take in a varaible called period that is some chrono duration